release/nightly_tests/dataset/map_benchmark.py

```diff
@@ -0,0 +1,87 @@
+import argparse
+
+import pyarrow as pa
+import pyarrow.compute as pc
+import pandas as pd
+import ray
+
+from benchmark import Benchmark
+
+
+def parse_args() -> argparse.Namespace:
+    parser = argparse.ArgumentParser(description="Map benchmark")
+    parser.add_argument(
+        "--api", choices=["map", "map_batches", "flat_map"], required=True
+    )
+    parser.add_argument(
+        "--sf", choices=["1", "10", "100", "1000", "10000"], default="1"
+    )
+    parser.add_argument("--batch-format", choices=["numpy", "pandas", "pyarrow"])
+    parser.add_argument("--compute", choices=["tasks", "actors"])
+    return parser.parse_args()
+
+
+def main(args: argparse.Namespace) -> None:
```

Comment:
Would it be possible to add verification of the records?

Reply:
I think it might be challenging.

We test correctness of these APIs in our unit tests, so I think it's okay if the performance tests don't also check correctness